### ISSUE TITLE
source code?

### ISSUE THREAD

**Issue by hx3k11** [MountainDrew8/Something#17]
hey can i get source code plz....

**Comment by TomyLobo** [comment]
+1, I would also like the source code

**Comment by Velosofy** [comment]
There is no way mods like Calamity will go open source

**Comment by TomyLobo** [comment]
yeah, I only wanted it so I can make a fork without some, let's say, controversial NPC names.
People want to play the game without thinking about politics, so I wanted them out.

But I found a way to patch the mod using [TML.Patcher](https://github.com/TML-Patcher/TML.Patcher) to extract the mod and Mono.Cecil to patch the names out of the binary.
Just a shame that I can't share that change with the world, cause that would be illegal.

**Comment by colegmr** [comment]
Ok ill bite I have to know 
what names are you talking about?
I read through all the calamity names and cant find anything
even remotely controversial.

**Comment by TomyLobo** [comment]
oh maybe they removed Ben Shapiro from the list by now, I haven't checked.
He was there in 0.11.8.5.

**Comment by colegmr** [comment]
oooo...
I think I remember that now that you mention it
But yeah hes gone and good riddance
the last thing we need is jews in our video games

Disclaimer:
(This is a sarcastic joke )^^^

seriously tho probably a good thing they took it out
Right or left real world politics don't belong anywhere near video games
and yea I doubt the source code thing will ever happen
anyway thanks for replying

**Comment by IDGCaptainRussia94** [comment]
Kinda makes you wonder what they contributed to the arctic code vault when they've never even used Github outside of getting icon uploading rights.

**Comment by lkyc2k** [comment]
I can't help but feel intentionally mislead since this is a website for OPEN SOURCE, and they distribute with both 'Source code.zip' and 'Source code.tar.gz', yet they are distributing PROPRIETARY software...

It feels like they're trying to trick users into believing the source code is being / can be audited by a third party.

**Comment by Velosofy** [comment]
Both of those archived "Source Code" are there by default by Github, sourcing from the repo.

I'm sure they made this just so they can have a place to host their mod files.

**Comment by IDGCaptainRussia94** [comment]
> 

as stated previously: mod icon, and also 50MB upload limit
Fab has no intention of using GitHub for the purpose it was meant for, the guy is far too paranoid of people plagiarizing his (even thou alot of it isn't his anymore) work.

...Even though most people who would do that don't use GitHub... Yeah.

**Comment by IDGCaptainRussia94** [comment]
> Both of those archived "Source Code" are there by default by Github, sourcing from the repo.
> 
> I'm sure they made this just so they can have a place to host their mod files.

> 

as stated previously: mod icon, and also 50MB upload limit
Fab has no intention of using GitHub for the purpose it was meant for, the guy is far too paranoid of people plagiarizing his (even thou alot of it isn't his anymore) work.

...Even though most people who would do that don't use GitHub... Yeah.

**Comment by IDGCaptainRussia94** [comment]
> I can't help but feel intentionally mislead since this is a website for OPEN SOURCE, and they distribute with both 'Source code.zip' and 'Source code.tar.gz', yet they are distributing PROPRIETARY software...
> 
> It feels like they're trying to trick users into believing the source code is being / can be audited by a third party.

Nah it's less complex than that, you see...

Terraria modding is still largely run by a bunch of children (IE people learn their 1st code here and many are under 18, very few experienced people come in from outside) who don't understand the concept of Open Sourcing and licensing, Hence why there's so many closed source (AKA PROPRIETARY) mods. People generally refuse to cooperate because everyone here has some kind of guts and glory popularity fetish, especially wanting to be like Calamity, and using libraries and dependencies makes you look weaker and less skilled because you didn't do it yourself.

As stated in my previous response: only reason this is here is because TML gives you enhanced uploading rights by linking a GitHub; they only did it for this purpose.

However because 1.4 mods are moving to the Steam Workshop I'd say this problem will fix itself and you'll stop seeing "open source but not really" mods on github.
